### PR TITLE
feat(docs-site): add doc for SGX Verifier.

### PR DESCRIPTION
**Pull request by swarna1101**

### CODE REVIEW

**Review by 0xjunger — APPROVED** [review]

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+title: SGXVerifier
+description: Taiko protocol page for "SGXVerifier.sol".
+---
+
+## Overview
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:6]:
```suggestion
```

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+The `SGXVerifier` smart contract implements SGX (Software Guard Extensions) signature proof verification on-chain. This verification ensures integrity and security of rollup state transitions by validating SGX-generated signatures. It also enables management and tracking of SGX instances through registration and replacement.
+
+---
+
+## Core Components
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:12]:
```suggestion
## Core Purpose
```
for consistency with taikol1-contract and taikol2-contract

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+### **SGX Instance Management**
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:14]:
```suggestion
**SGX Instance Management**
```

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+- **Instance Registry**:
+
+  - Each SGX instance is uniquely identified by its Ethereum address (derived from an ECDSA public-private key pair generated in the SGX enclave).
+  - The registry ensures:
+    - Only valid instances are allowed.
+    - Instances are valid for a predefined duration (`INSTANCE_EXPIRY`).
+
+- **Instance Lifecycle**:
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:23]:
```suggestion
1. **Instance Registry**:

  - Each SGX instance is uniquely identified by its Ethereum address (derived from an ECDSA public-private key pair generated in the SGX enclave).
  - The registry ensures:
    - Only valid instances are allowed.
    - Instances are valid for a predefined duration (`INSTANCE_EXPIRY`).

2.  **Instance Lifecycle**:
```

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  - **Addition**: SGX instances can be added via the `addInstances` function or the `registerInstance` method (following attestation verification).
+  - **Replacement**: Old SGX instances can be replaced with new ones to maintain security.
+  - **Deletion**: Instances can be removed using the `deleteInstances` function.
+
+---
+
+## Functions
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:30]:
```suggestion
## Key Functions
```

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+### **`addInstances`**
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:32]:
```suggestion
### `addInstances`
```

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+- **Purpose**: Adds new SGX instances to the registry.
+- **Input**:
+  - `_instances`: Array of Ethereum addresses corresponding to the SGX instances.
+- **Output**: Returns an array of assigned instance IDs.
+- **Access Control**: Restricted to the owner.
+
+---
+
+### **`deleteInstances`**
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:42]:
```suggestion
### `deleteInstances`
```

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+- **Purpose**: Removes SGX instances from the registry.
+- **Input**:
+  - `_ids`: Array of instance IDs to be removed.
+- **Access Control**: Restricted to the owner or the `SGX_WATCHDOG` role.
+
+---
+
+### **`registerInstance`**
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:51]:
```suggestion
### `registerInstance`
```

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+- **Purpose**: Registers an SGX instance by verifying its attestation off-chain and adding it to the registry.
+- **Input**:
+  - `_attestation`: Parsed attestation quote containing SGX enclave report details.
+- **Output**: Returns the assigned instance ID.
+- **Access Control**: Open to external calls.
+
+---
+
+### **`verifyProof`**
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:61]:
```suggestion
### `verifyProof`
```

File: packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+  - `_ctx`: Context of the proof.
+  - `_tran`: Transition data.
+  - `_proof`: SGX signature proof.
+- **Mechanism**:
+  - Validates the instance ID and signature.
+  - Ensures the SGX instance is valid and replaces it if needed.
+
+---
+
+### **`verifyBatchProof`**
```
Comment (RogerLamTd) [inline packages/docs-site/src/content/docs/taiko-protocol/codebase-analysis/sgxverifier-contract.md:74]:
```suggestion
### `verifyBatchProof`
```

**Review by RogerLamTd — APPROVED** [review]

**Pull request merged**